riemann/serializer.js

```diff
@@ -2,50 +2,74 @@
    and cache it in memory. */
 var riemannSchema;
 if (!riemannSchema) {
-  var Schema    = require('node-protobuf');
-  var readFile  = require('fs').readFileSync;
-  riemannSchema = new Schema(readFile(__dirname+'/proto/proto.desc'));
+  var protobuf = require('protobufjs')
+
+  protobuf.load(__dirname + '/proto/proto.proto', function (err, root) {
+    if (err) throw err
+
+    // Pull the message type out.
+    riemannSchema = root
+  })
 }
 
-function _serialize(type, value) {
-  return riemannSchema.serialize(value, type);
+function _serialize (type, value) {
+  var messageType = riemannSchema.lookupType(type)
+
+  // https://www.npmjs.com/package/protobufjs#valid-message
+  var errorString = messageType.verify(value)
+  var message
+
+  // Using create is faster, so only fall back to fromObject in worst case.
+  if (typeof errorString === 'string') {
+    message = messageType.fromObject(value)
+  } else { // It will be null here.
+    message = messageType.create(value)
+  }
+
+  var buffer = messageType.encode(message).finish()
```

Comment:
I believe you could simply return the result of finish() instead of assignment here.